### PR TITLE
[SPARK-30993][FOLLOWUP][SQL] Refactor LocalDateTimeUDT as YearUDT in UserDefinedTypeSuite

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Refactor LocalDateTimeUDT as YearUDT in UserDefinedTypeSuite

### Why are the changes needed?

As we are going to support java.time.LocalDateTime as an external type of TimestampWithoutTZ type https://github.com/apache/spark/pull/32814, registering java.time.LocalDateTime as UDT will cause test failures: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/139469/testReport/
This PR is to unblock https://github.com/apache/spark/pull/32814.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Unit test.

### CODE REVIEW

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/UserDefinedTypeSuite.scala [hunk 1]
```diff
@@ -258,19 +258,17 @@ class UserDefinedTypeSuite extends QueryTest with SharedSparkSession with Parque
 
   test("SPARK-30993: UserDefinedType matched to fixed length SQL type shouldn't be corrupted") {
     def concatFoo(a: FooWithDate, b: FooWithDate): FooWithDate = {
-      FooWithDate(b.date, a.s + b.s, a.i)
+      FooWithDate(b.year, a.s + b.s, a.i)
     }
 
-    UDTRegistration.register(classOf[LocalDateTime].getName, classOf[LocalDateTimeUDT].getName)
+    UDTRegistration.register(classOf[Year].getName, classOf[YearUDT].getName)
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/UserDefinedTypeSuite.scala:264]: shall we unregister it after test finishes?

Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/UserDefinedTypeSuite.scala:264]: oh there is no such API yet...

**Comment by gengliangwang** [comment]
Merging to master

**Comment by dongjoon-hyun** [comment]
Hi, @gengliangwang . 
SPARK-30993 was released long time ago by 2.4.6 and 3.0.0.
If you don't mind, could you use a new JIRA instead of the follow-up next time?

I'm not aware of the community rule for this. Usually, I use a follow-up when it will have the same release version which means the original patch is not released yet.

**Comment by gengliangwang** [comment]
@dongjoon-hyun Yes, I think creating a new Jira makes more sense. Thanks for the suggestion!

**Pull request closed without merging**